gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageGrpcIntegrationTest.java

```diff
@@ -0,0 +1,193 @@
+
+    GoogleCloudStorageItemInfo overwrittenItemInfo = rawStorage.getItemInfo(objectToCreate);
+    assertThat(overwrittenItemInfo.exists()).isTrue();
+    assertThat(overwrittenItemInfo.getSize()).isEqualTo(256);
+    assertObjectContent(rawStorage, objectToCreate, overwriteBytesToWrite);
+  }
+
+  @Test
+  public void testCreateEmptyObject() throws IOException {
+    GoogleCloudStorage rawStorage = createGoogleCloudStorage();
+    StorageResourceId objectToCreate =
+        new StorageResourceId(BUCKET_NAME, "testCreateEmptyObject_Object");
+
+    rawStorage.createEmptyObject(objectToCreate);
+
+    GoogleCloudStorageItemInfo itemInfo = rawStorage.getItemInfo(objectToCreate);
+
+    assertThat(itemInfo.exists()).isTrue();
+    assertThat(itemInfo.getSize()).isEqualTo(0);
+  }
+
+  @Test
+  public void testCreateInvalidObject() throws IOException {
+    GoogleCloudStorage rawStorage = createGoogleCloudStorage();
+    StorageResourceId objectToCreate =
+        new StorageResourceId(BUCKET_NAME, "testCreateInvalidObject_InvalidObject\n");
+
+    assertThrows(
+        IOException.class, () -> writeObject(rawStorage, objectToCreate, /* objectSize= */ 10));
+  }
+
+  @Test
+  public void testOpen() throws IOException {
+    GoogleCloudStorage rawStorage = createGoogleCloudStorage();
+    StorageResourceId objectToCreate = new StorageResourceId(BUCKET_NAME, "testOpen_Object");
+    byte[] objectBytes = writeObject(rawStorage, objectToCreate, /* objectSize= */ 100);
+
+    assertObjectContent(rawStorage, objectToCreate, objectBytes);
+  }
+
+  @Test
+  public void testOpenNonExistentItem() throws IOException {
+    GoogleCloudStorage rawStorage = createGoogleCloudStorage();
+    assertThrows(
+        FileNotFoundException.class,
+        () ->
+            rawStorage.open(new StorageResourceId(BUCKET_NAME, "testOpenNonExistentItem_Object")));
+  }
+
+  @Test
+  public void testOpenEmptyObject() throws IOException {
+    GoogleCloudStorage rawStorage = createGoogleCloudStorage();
+    StorageResourceId resourceId = new StorageResourceId(BUCKET_NAME, "testOpenEmptyObject_Object");
+    rawStorage.createEmptyObject(resourceId);
+
+    assertObjectContent(rawStorage, resourceId, new byte[0]);
+  }
+
+  @Test
+  public void testOpenLargeObject() throws IOException {
+    GoogleCloudStorage rawStorage = createGoogleCloudStorage();
+    StorageResourceId resourceId = new StorageResourceId(BUCKET_NAME, "testOpenLargeObject_Object");
+
+    int partitionsCount = 50;
+    byte[] partition =
+        writeObject(rawStorage, resourceId, /* partitionSize= */ 10 * 1024 * 1024, partitionsCount);
+
+    assertObjectContent(rawStorage, resourceId, partition, partitionsCount);
+  }
+
+  @Test
+  public void testPartialRead() throws IOException {
+    GoogleCloudStorage rawStorage = createGoogleCloudStorage();
+    final int segmentSize = 553;
+    final int segmentCount = 5;
```

Comment:
`final` modifiers for local variables are redundant.

Reply:
Done.